modules/rte_mis_state/rte_mis_state.module

```diff
@@ -757,101 +760,85 @@ function rte_mis_state_user_presave(UserInterface $account) {
 }
 
 /**
- * Implements hook_entity_presave().
+ * Implements hook_workflow().
  */
-function rte_mis_state_mini_node_presave(EntityInterface $entity) {
-
-  if ($entity->bundle() === 'school_details') {
-
+function rte_mis_state_workflow($op, WorkflowTransitionInterface $transition, UserInterface $user) {
+  if ($op == 'transition pre' && $transition->wid->getString() == 'school_registration_verification' && $transition->getTargetEntityTypeId() == 'mini_node' && $transition->hasStateChange()) {
```

Comment:
Did we checked the existing functionality which are getting effected with this?

Reply:
yes,this was used to send sms and email for the state change of school application. I have tested this by sending mail and SMS.
![image](https://github.com/Indus-Action-Initiatives/rte-mis/assets/124239255/06bf986a-e597-4cc2-ae16-ae4f71ccae69)

Reply:
@RohanSingh-inno We have removed some piece of code as well. Any specific reason for that?

Reply:
yes, we have used a more optimized way of doing the above task. Before we were loading the taxonomy to load the user data, but it could be accessed directly as we have an entity reference field for the school mini node in the user entity.